### PR TITLE
Add git-runic script

### PR DESCRIPTION
**Pull request by fredrikekre**
Adds the [`git-runic`](https://github.com/fredrikekre/Runic.jl/blob/master/bin/git-runic) script (a modified version of [`git-clang-format`](https://github.com/llvm/llvm-project/blob/main/clang/tools/clang-format/git-clang-format)) from https://github.com/fredrikekre/Runic.jl/pull/120#issuecomment-2522991937 to the repository.

### CODE REVIEW

**Comment by fredrikekre** [comment]
cc @maleadt

File: README.md [hunk 1]
```diff
@@ -267,9 +271,28 @@ This assumes that Runic is installed in the `@runic` shared project as suggested
...
+provides a convenient way to apply Runic formatting incrementally to a code
+base by limiting formatting to lines that are added or modified in each commit.
+The script can be installed with the following commands (replace the two
+occurences of `~/.local/bin` if needed):
+
+```sh
+# Download the script into ~/.local/bin
+curl -fsSL -o ~/.local/bin/git-runic https://raw.githubusercontent.com/fredrikekre/Runic.jl/refs/heads/master/bin/git-runic
+# Make the script executable
+chmod +x ~/.local/bin/git-runic
```
Comment (maleadt) [inline README.md:288]: Duplicated with the installation instructions at the top?

**Review by maleadt — COMMENTED** [review]
LGTM! Has been working great locally; thanks for adding the necessary features.

File: README.md (hunk 1 above)
Comment (fredrikekre) [reply README.md:288]: That was on purpose. The instructions at the top are just a quick copy-paste thing but this here has some more details.

**Comment by fredrikekre** [comment]
Do you automatically run `git runic` in a pre-commit hook or similar? Or just manually before committing?

**Comment by maleadt** [comment]
I personally do it manually, after staging changes, because I want to be able to choose which changes I stage.

**Comment by fredrikekre** [comment]
Ha, okay. Just out of curiosity -- what formatting don't you like?

**Comment by maleadt** [comment]
I generally like the formatting, but sometimes I've carefully laid out values for clarify, e.g.:
- https://github.com/JuliaGPU/CUDA.jl/blob/4e9513b8a4e56629a236b58504d609b1775a8236/src/device/texture.jl#L58-L61
- https://github.com/JuliaGPU/CUDA.jl/blob/4e9513b8a4e56629a236b58504d609b1775a8236/src/device/texture.jl#L169-L172

These become:

```patch
@@ -56,31 +56,45 @@ end
 Base.Tuple(x::Vec4) = tuple(x.a, x.b, x.c, x.d)

 for (dispatch_rettyp, julia_rettyp, llvm_rettyp) in
-        ((:Signed,        :(Vec4{Int32}),   :v4u32),
-         (:Unsigned,      :(Vec4{UInt32}),  :v4s32),
-         (:AbstractFloat, :(Vec4{Float32}), :v4f32))
+        (
+        (:Signed, :(Vec4{Int32}), :v4u32),
+        (:Unsigned, :(Vec4{UInt32}), :v4s32),
+        (:AbstractFloat, :(Vec4{Float32}), :v4f32),
+    )

@@ -166,9 +184,13 @@ end
     h0y = h0(fy)
     h1y = h1(fy)

-    vals = g0(fy) .* (g0x .* tex(t, px + h0x, py + h0y) .+
-                      g1x .* tex(t, px + h1x, py + h0y)) .+
-           g1(fy) .* (g0x .* tex(t, px + h0x, py + h1y) .+
-                      g1x .* tex(t, px + h1x, py + h1y))
+    vals = g0(fy) .* (
+        g0x .* tex(t, px + h0x, py + h0y) .+
+            g1x .* tex(t, px + h1x, py + h0y)
+    ) .+
+        g1(fy) .* (
+        g0x .* tex(t, px + h0x, py + h1y) .+
+            g1x .* tex(t, px + h1x, py + h1y)
+    )
     return (unpack(T, vals))
 end
```

It seems silly to annotate those with `# RUNIC NO TOUCHY` (or whatever the magical comment is), so I think I prefer the manual approach instead.

**Pull request merged**